modules/apm/src/main/plugin-metadata/plugin-security.policy

```diff
@@ -19,4 +19,11 @@ grant codeBase "${codebase.elastic-apm-agent}" {
     permission java.lang.RuntimePermission "setFactory";
     permission java.lang.reflect.ReflectPermission "suppressAccessChecks";
     permission java.net.SocketPermission "*", "connect,resolve";
+    // profiling function in APM agent
+    permission java.util.PropertyPermission "AsyncProfiler.safemode", "write";
+    permission java.lang.RuntimePermission "accessUserInformation";
+    permission java.lang.RuntimePermission "loadLibrary.*";
+    permission java.lang.RuntimePermission "getClassLoader";
+    permission java.io.FilePermission "<<ALL FILES>>", "read,write";
+    permission org.elasticsearch.secure_sm.ThreadPermission "modifyArbitraryThreadGroup";
```

Comment:
added due to the following error:
```
2022-09-30 20:33:32,521 [elastic-apm-sampling-profiler] ERROR co.elastic.apm.agent.profiler.SamplingProfiler - Stopping profiler
java.security.AccessControlException: access denied ("org.elasticsearch.secure_sm.ThreadPermission" "modifyArbitraryThreadGroup")
        at java.security.AccessControlContext.checkPermission(AccessControlContext.java:485) ~[?:?]
        at java.security.AccessController.checkPermission(AccessController.java:1068) ~[?:?]
        at java.lang.SecurityManager.checkPermission(SecurityManager.java:411) ~[?:?]
        at org.elasticsearch.secure_sm.SecureSM.checkThreadGroupAccess(SecureSM.java:195) ~[elasticsearch-secure-sm-8.5.0-SNAPSHOT.jar:?]
        at org.elasticsearch.secure_sm.SecureSM.checkAccess(SecureSM.java:135) ~[elasticsearch-secure-sm-8.5.0-SNAPSHOT.jar:?]
        at java.lang.ThreadGroup.checkAccess(ThreadGroup.java:335) ~[?:?]
        at java.lang.ThreadGroup.enumerate(ThreadGroup.java:441) ~[?:?]
        at co.elastic.apm.agent.profiler.ThreadMatcher.forEachThread(ThreadMatcher.java:41) ~[?:?]
        at co.elastic.apm.agent.profiler.SamplingProfiler.restoreFilterState(SamplingProfiler.java:433) ~[?:?]
        at co.elastic.apm.agent.profiler.SamplingProfiler.profile(SamplingProfiler.java:402) ~[?:?]
        at co.elastic.apm.agent.profiler.SamplingProfiler.run(SamplingProfiler.java:377) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

Required additionally permitting this to the "class permissions" for modules. In actuality, this permission is preventing the ProfilingSampler from doing [ThreadGroup::enumerate](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/ThreadGroup.html#enumerate(java.lang.Thread%5B%5D)) in order to pass threads to async-profiler, not actually working to modify them. Anyway it doesn't seem so concerning as the already-permitted-for-modules `<<ALL FILES>> | read,write` so I've added it